pkg/build/strategies/sti/sti.go

```diff
@@ -323,15 +324,19 @@ func (b *STI) Exists(config *api.Config) bool {
 		return false
 	}
 
-	// can only do incremental build if runtime image exists, so always pull image
-	previousImageExists, _ := b.docker.IsImageInLocalRegistry(config.Tag)
```

Comment:
you've broken this behavior which was by design to pull the previous version of the application image if it didn't exist, regardless of the "force pull" flag.  

whether or not we pull the previous application image should be entirely independent of the BuilderPullPolicy.

The right way to fix this is to go ahead and introduce a PreviousImagePullPolicy.